paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java

```diff
@@ -27,30 +27,40 @@
 import org.apache.paimon.utils.SnapshotManager;
 import org.apache.paimon.utils.TagManager;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import javax.annotation.Nullable;
 
 import java.time.Duration;
 import java.time.Instant;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.SortedMap;
 
 import static org.apache.paimon.Snapshot.FIRST_SNAPSHOT_ID;
 import static org.apache.paimon.shade.guava30.com.google.common.base.MoreObjects.firstNonNull;
+import static org.apache.paimon.utils.Preconditions.checkArgument;
 import static org.apache.paimon.utils.Preconditions.checkState;
 
 /** A manager to create tags automatically. */
 public class TagAutoCreation {
```

Comment:
Only creating this class when table has auto tag creation options.

We need to create another class? Or create a `TagAutoManager` to wrap `TagAutoCreation` and `TagAutoExpire`.

Reply:
done
I create a TagAutoManager to wrap TagAutoCreation and TagAutoExpire.